hudi-flink-datasource/hudi-flink1.14.x/src/main/java/org/apache/hudi/table/format/cow/ParquetSplitReaderUtil.java

```diff
@@ -119,6 +119,11 @@ public static ParquetColumnarRowSplitReader genPartColumnarRowReader(
       long splitLength,
       FilterPredicate filterPredicate,
       UnboundRecordFilter recordFilter) throws IOException {
+
+    if (Arrays.stream(selectedFields).anyMatch(x -> x == -1)) {
+      throw new AssertionError("One or more specified columns does not exist in the hudi table");
```

Comment:
Maybe you can just use `ValidationUtils.checkState`.

Reply:
have updated it to `ValidationUtils.checkState`